### PR TITLE
Remove deprecation warnings by switching MinimalLib to fingerprint generators

### PR DESCRIPTION
**Pull request by ptosco**
This PR also contains small cleanup to MinimalLib `CMakeLists.txt`.

### CODE REVIEW

File: Code/MinimalLib/common.h [hunk 1]
```diff
@@ -8,7 +8,9 @@
 //  of the RDKit source tree.
 //
 #pragma once
+#ifdef __GNUC__
```
Comment (greglandrum) [inline Code/MinimalLib/common.h:11]: If this is going to stay here, and I don't think it needs to after these changes are made, we probably also want it one for clang, right?

**Review by greglandrum — COMMENTED** [review]
One question

File: Code/MinimalLib/common.h (hunk 1 above)
Comment (ptosco) [reply Code/MinimalLib/common.h:11]: You added that pragma on 21/12/2023, I think to make CI happy. I only added the `#ifdef` guard to silence the "unknown pragma" warning in emscripten builds. It ended up in this PR but is not really connected to the fingerprint work.

**Review by greglandrum — APPROVED** [review]
LGTM

**Pull request merged**